### PR TITLE
Improve Notification Area settings

### PR DESCRIPTION
**Pull request by furgo16**
Fixes: #15149

Incremental improvements to the Notification Area settings dialog:
1. Reorganize settings into checkable groups that automatically hide their children settings for improved usabilty.
1. Remove unnecessary code to hide/unhide widgets (the checkable groups replace its need).
1. Write setting labels consistently with sentence case.
1. Rephrase/expand tooltips to improve clarity and to use consistent nomenclature.
1. Renamed non-intrusive notifications label to [pop-up notifications](https://en.wikipedia.org/wiki/Pop-up_notification) to use standard nomenclature.
1. Add unit suffixes (e.g. ` px`) to units in input boxes.

| | Disabled notification area | Enabled notification area | Enabled notification area and pop-up notifications |
| --- | --- | --- | --- |
| Before | Captura de pantalla de 2024-07-05 11-22-22  | Captura de pantalla de 2024-07-05 11-22-30 | Captura de pantalla de 2024-07-05 11-22-37 |
| After | Captura de pantalla de 2024-07-05 11-04-12 | Captura de pantalla de 2024-07-05 11-04-17 | Captura de pantalla de 2024-07-05 11-04-32 |

At the time of submission, FreeCAD 1.0 is in User Interface Freeze. This PR is originally intended to be merged after 1.0. However, if there is a User Interface Freeze exception process, a case could be made to justify its inclusion in 1.0.

Happy to add unit tests too if needed (but would need some guidance on their implementation for FreeCAD).

### CODE REVIEW

**Comment by PaddleStroke** [comment]
Nice reorganization.

File: src/Gui/PreferencePages/DlgSettingsNotificationArea.cpp [hunk 1]
```diff
@@ -55,8 +56,17 @@ DlgSettingsNotificationArea::~DlgSettingsNotificationArea() = default;
...
-    ui->NonIntrusiveNotificationsEnabled->onSave();
+    // must be done as very first because we create a new instance of NavigatorStyle
+    // where we set some attributes afterwards
+    ParameterGrp::handle hGrp = App::GetApplication().GetParameterGroupByPath
+        ("User parameter:BaseApp/Preferences/NotificationArea");
+
+    bool isNotificationAreaEnabled = ui->NotificationAreaEnabled->isChecked();
+    hGrp->SetBool("NotificationAreaEnabled", isNotificationAreaEnabled);
+    
+    bool isNonIntrusiveNotificationsEnabled = ui->NotificationAreaEnabled->isChecked();
```
Comment (luzpaz) [inline src/Gui/PreferencePages/DlgSettingsNotificationArea.cpp:67]: Fix trailing whitespace on line 66

File: src/Gui/PreferencePages/DlgSettingsNotificationArea.cpp [hunk 2]
```diff
@@ -55,8 +56,17 @@ DlgSettingsNotificationArea::~DlgSettingsNotificationArea() = default;
...
-    ui->NotificationAreaEnabled->onSave();
-    ui->NonIntrusiveNotificationsEnabled->onSave();
+    // must be done as very first because we create a new instance of NavigatorStyle
+    // where we set some attributes afterwards
+    ParameterGrp::handle hGrp = App::GetApplication().GetParameterGroupByPath
+        ("User parameter:BaseApp/Preferences/NotificationArea");
+
+    bool isNotificationAreaEnabled = ui->NotificationAreaEnabled->isChecked();
+    hGrp->SetBool("NotificationAreaEnabled", isNotificationAreaEnabled);
+
```
Comment (furgo16) [inline src/Gui/PreferencePages/DlgSettingsNotificationArea.cpp:66]:
```suggestion

```

File: src/Gui/PreferencePages/DlgSettingsNotificationArea.cpp (hunk 1 above)
Comment (furgo16) [reply src/Gui/PreferencePages/DlgSettingsNotificationArea.cpp]: Fixed, thanks. Feel free to use the "Suggestion" button in GitHub in the future. It's quicker for the reviewer to do it than commenting, and the author can just apply it at the click of a button. Thank you.

**Comment by maxwxyz** [comment]
@furgo16 can you please resolve the conflicts?
@FreeCAD/design-working-group is this good to go?

**Comment by furgo16** [comment]
Oh wow, nice to see all post-1.0 PRs picked up. This morning is like Christmas for main branch users! :-)

Fixing conflicts now.

**Comment by furgo16** [comment]
@maxwxyz conflict resolved, thanks for the heads up

**Comment by maxwxyz** [comment]
looks good!

**Comment by chennes** [comment]
@obelisk79 can you all discuss this and let us know if the DWG is OK with the change?

**Comment by PaddleStroke** [comment]
@furgo16 can you please solve the conflict?

**Comment by maxwxyz** [comment]
@furgo16 CI is failing, could you please take a look at this?

**Comment by furgo16** [comment]
Thanks for the heads up. I can't look at it right now, but I'll try this afternoon. At a quick glance, the error is:

```
uic: Error in line 38, column 32 : Unexpected element property
File '/home/runner/work/FreeCAD/FreeCAD/src/Gui/PreferencePages/DlgSettingsNotificationArea.ui' is not valid
```

But for some reason the GitHub app is not loading the file on my phone, so I can't tell if it's a trivial fix or not.

File: src/Gui/PreferencePages/DlgSettingsNotificationArea.ui [hunk 3]
```diff
@@ -7,209 +7,34 @@
...
-      <item row="3" column="1">
-       <widget class="Gui::PrefSpinBox" name="notificationWidth">
-        <property name="toolTip">
-         <string>Width of the notification in pixels</string>
+         <string>If checked, non-intrusive pop-up notifications will appear above the status bar, around the notification area. Pop-up notifications can be manually dismissed by clicking on them, and also automatically dismissed by specifying a maximum and minimum duration for them to be displayed.
+
+Additionally, pop-up notifications can be disabled. In this case the user can still use the notification area as a quick-access location to view notifications, without the distracton of an additional pop-up.</string>
         </property>
+        <property name="title">
+         <string>Enable pop-up notifications</string>
```
Comment (furgo16) [inline src/Gui/PreferencePages/DlgSettingsNotificationArea.ui:37]:
```suggestion
         <string>Enable pop-up notifications</string>
        </property>
```

Fix merge error

**Comment by PaddleStroke** [comment]
@furgo16 I suggest perhaps dropping https://github.com/FreeCAD/FreeCAD/commit/11d3008046a286981a0fd27bc850db6f012b2706 and the following commits. Then rebase on upstream main.

**Comment by PaddleStroke** [comment]
Note I have checked and now the .ui file is identical to the one before https://github.com/FreeCAD/FreeCAD/commit/11d3008046a286981a0fd27bc850db6f012b2706

So I think you fixed the issue.

**Pull request merged**